llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Passes/TileAndDecomposeWinogradPass.cpp

```
@@ -375,19 +508,39 @@ struct TileAndDecomposeWinogradTransformPass
         linalg::LinalgDialect, scf::SCFDialect, tensor::TensorDialect>();
   }
 
+  TileAndDecomposeWinogradTransformPass() = default;
+  TileAndDecomposeWinogradTransformPass(bool onlyTile) {
+    this->onlyTile = onlyTile;
+  }
+  TileAndDecomposeWinogradTransformPass(
+      const TileAndDecomposeWinogradTransformPass &pass) {
+    onlyTile = pass.onlyTile;
+  }
+
   void runOnOperation() override;
 };
 } // namespace
 
+LogicalResult reifyWinogradTransform(func::FuncOp funcOp, bool onlyTile) {
+  IRRewriter rewriter(funcOp.getContext());
+  funcOp.walk([&](WinogradInputTransformOp inputOp) {
```

If any one of these fail shouldn't we return a failure?

---

Hi @harsh-nod you're correct - I've addressed this in the latest revision I pushed in this PR. You may take a look.
So, I'm now simply capturing the `if any` clause but not making an early exit - this is to capture as many rewriting errors as we can with a single run.